persistence/trees/trees.go

```diff
@@ -90,6 +91,8 @@ const (
 // of the underlying trees by utilizing the lazy loading
 // functionality provided by the underlying smt library.
 type treeStore struct {
+	modules.Module
```

Comment:
Add type assertion to guarantee that `treeStore` implements the module interface like we do with `var _ = `

Reply:
Also, embed the appropriate factory interface we use introduced by Bryan

Reply:
Yessir

Reply:
Done 👍